_posts/2019-03-19-til-validations-and-constraints.md

```diff
@@ -0,0 +1,229 @@
+---
+author: Prince Wilson
+date: 2019-03-19
+layout: post
+categories: til
+author_link: https://github.com/maxcell
+title:  TIL Constraints and Validations
+excerpt: >
+  Let's take a look at how Ecto handles these two ways
+  of ensuring data integrity
+---
+
+Developers want to create the best applications they can for their users.
+In the process, they want to make sure to give good feedback to their users
+when data doesn't get saved into the database. In Elixir, there is a great tool
+on top of the database that helps, Ecto! It can put validations and
+constraints onto specific fields to ensure data integrity.
+
+However, did you know they are differences between validations and constraints?
+I didn't. In the process of building a side project, I ran into the problem a few
+times! Let's talk about the goal of them and see the differences. We'll dive into
```

Comment:
Might be more clear to say "Let's talk about the purpose of each approach and the differences between them"